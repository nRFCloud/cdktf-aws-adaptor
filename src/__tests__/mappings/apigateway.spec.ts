import { ApiGatewayAccount } from "@cdktf/provider-aws/lib/api-gateway-account/index.js";
import { ApiGatewayBasePathMapping } from "@cdktf/provider-aws/lib/api-gateway-base-path-mapping/index.js";
import { Apigatewayv2Authorizer } from "@cdktf/provider-aws/lib/apigatewayv2-authorizer/index.js";
import { Apigatewayv2DomainName } from "@cdktf/provider-aws/lib/apigatewayv2-domain-name/index.js";
import { Apigatewayv2Integration } from "@cdktf/provider-aws/lib/apigatewayv2-integration/index.js";
import { Apigatewayv2Stage } from "@cdktf/provider-aws/lib/apigatewayv2-stage/index.js";
import { CloudcontrolapiResource } from "@cdktf/provider-aws/lib/cloudcontrolapi-resource/index.js";
import { CfnAccount, CfnBasePathMapping, CfnDeployment, CfnStage } from "aws-cdk-lib/aws-apigateway";
import {
  CfnAuthorizer as CfnAuthorizerV2,
  CfnDomainName as CfnDomainNameV2,
  CfnIntegration as CfnIntegrationV2,
  CfnStage as CfnStageV2,
} from "aws-cdk-lib/aws-apigatewayv2";
import { setupJest } from "cdktf/lib/testing/adapters/jest.js";
import { registerMappings } from "../../mappings/index.js";
import { itShouldMapCfnElementToTerraformResource, synthesizeElementAndTestStability } from "../helpers.js";

setupJest();
registerMappings();

describe("Apigateway mappings", () => {
  it("should map AWS::ApiGateway::Account", () => {
    synthesizeElementAndTestStability(
      CfnAccount,
      {
        cloudWatchRoleArn: "cw-role-arn",
      },
      ApiGatewayAccount,
      {
        cloudwatchRoleArn: "cw-role-arn",
      },
    );
  });

  it("should map AWS::ApiGateway::Deployment", () => {
    synthesizeElementAndTestStability(
      CfnDeployment,
      {
        restApiId: "rest-api-id",
        stageName: "stage-name",
        description: "description",
        stageDescription: {
          creationStack: ["stack"],
          cacheClusterEnabled: true,
          cacheClusterSize: "cache-cluster-size",
          canarySetting: {
            percentTraffic: 50,
            stageVariableOverrides: {
              "stage-variable": "stage-variable-value",
            },
          },
          accessLogSetting: {
            destinationArn: "destination-arn",
          },
          cacheDataEncrypted: true,
          cacheTtlInSeconds: 60,
          cachingEnabled: true,
          clientCertificateId: "client-certificate-id",
          dataTraceEnabled: true,
          loggingLevel: "logging-level",
          metricsEnabled: true,
          documentationVersion: "documentation-version",
          tracingEnabled: true,
          methodSettings: [
            {
              dataTraceEnabled: true,
              httpMethod: "http-method",
              loggingLevel: "logging-level",
              metricsEnabled: true,
              resourcePath: "resource-path",
              throttlingBurstLimit: 60,
              throttlingRateLimit: 60,
              cachingEnabled: true,
              cacheDataEncrypted: true,
              cacheTtlInSeconds: 60,
            },
          ],
          throttlingBurstLimit: 60,
          throttlingRateLimit: 60,
        },
        deploymentCanarySettings: {
          percentTraffic: 50,
          stageVariableOverrides: {
            "stage-variable": "stage-variable-value",
          },
          creationStack: ["stack"],
          useStageCache: true,
        },
      },
      CloudcontrolapiResource,
      {
        typeName: "AWS::ApiGateway::Deployment",
        desiredState: expect.stringContaining(""),
      },
    );
  });

  it("should map AWS::ApiGateway::Stage", () => {
    synthesizeElementAndTestStability(
      CfnStage,
      {
        accessLogSetting: {
          destinationArn: "destination-arn",
        },
        cacheClusterEnabled: true,
        cacheClusterSize: "cache-cluster-size",
        canarySetting: {
          percentTraffic: 50,
          stageVariableOverrides: {
            "stage-variable": "stage-variable-value",
          },
        },
        clientCertificateId: "client-certificate-id",
        stageName: "stage-name",
        description: "description",
        documentationVersion: "documentation-version",
        tracingEnabled: true,
        methodSettings: [],
        restApiId: "rest-api-id",
        deploymentId: "deployment-id",
      },
      CloudcontrolapiResource,
      {
        typeName: "AWS::ApiGateway::Stage",
        desiredState: expect.stringContaining(""),
      },
    );
  });

  it("should map AWS::ApiGateway::BasePathMapping", () => {
    synthesizeElementAndTestStability(
      CfnBasePathMapping,
      {
        restApiId: "rest-api-id",
        basePath: "base-path",
        domainName: "domain-name",
        stage: "stage",
        id: "id",
      },
      ApiGatewayBasePathMapping,
      {
        apiId: "rest-api-id",
        basePath: "base-path",
        domainName: "domain-name",
        stageName: "stage",
        id: "id",
      },
    );
  });

  itShouldMapCfnElementToTerraformResource(
    CfnStageV2,
    {
      accessLogSettings: {
        destinationArn: "destination-arn",
        format: "format",
      },
      apiId: "api-id",
      autoDeploy: true,
      clientCertificateId: "client-certificate-id",
      defaultRouteSettings: {
        dataTraceEnabled: true,
        detailedMetricsEnabled: true,
        loggingLevel: "logging-level",
        throttlingBurstLimit: 60,
        throttlingRateLimit: 60,
      },
      deploymentId: "deployment-id",
      description: "description",
      stageVariables: {
        "stage-variable": "stage-variable-value",
      },
      stageName: "stage-name",
      routeSettings: {
        DataTraceEnabled: true,
        RouteKey: "route-key",
      },
      tags: {
        "tag": "tag-value",
      },
    },
    Apigatewayv2Stage,
    {
      routeSettings: [{
        dataTraceEnabled: true,
        routeKey: "route-key",
      }],
      tags: {
        "tag": "tag-value",
      },
      accessLogSettings: {
        destinationArn: "destination-arn",
        format: "format",
      },
      apiId: "api-id",
      autoDeploy: true,
      description: "description",
      name: "stage-name",
      stageVariables: {
        "stage-variable": "stage-variable-value",
      },
      deploymentId: "deployment-id",
      clientCertificateId: "client-certificate-id",
      defaultRouteSettings: {
        dataTraceEnabled: true,
        detailedMetricsEnabled: true,
        loggingLevel: "logging-level",
        throttlingBurstLimit: 60,
        throttlingRateLimit: 60,
      },
    },
  );

  itShouldMapCfnElementToTerraformResource(
    CfnDomainNameV2,
    {
      domainName: "domain-name",
      domainNameConfigurations: [{
        certificateArn: "certificate-arn",
        endpointType: "REGIONAL",
        securityPolicy: "TLS_1_2",
        certificateName: "certificate-name",
        ownershipVerificationCertificateArn: "ownership-verification-certificate-arn",
      }],
      mutualTlsAuthentication: {
        truststoreUri: "truststore-uri",
        truststoreVersion: "truststore-version",
      },
      tags: {
        "tag": "tag-value",
      },
    },
    Apigatewayv2DomainName,
    {
      tags: {
        "tag": "tag-value",
      },
      domainName: "domain-name",
      mutualTlsAuthentication: {
        truststoreUri: "truststore-uri",
        truststoreVersion: "truststore-version",
      },
      domainNameConfiguration: {
        certificateArn: "certificate-arn",
        endpointType: "REGIONAL",
        securityPolicy: "TLS_1_2",
        ownershipVerificationCertificateArn: "ownership-verification-certificate-arn",
      },
    },
  );

  itShouldMapCfnElementToTerraformResource(
    CfnIntegrationV2,
    {
      apiId: "api-id",
      connectionId: "connection-id",
      connectionType: "connection-type",
      contentHandlingStrategy: "content-handling-strategy",
      credentialsArn: "credentials-arn",
      description: "description",
      integrationMethod: "integration-method",
      integrationSubtype: "integration-subtype",
      integrationType: "integration-type",
      integrationUri: "integration-uri",
      passthroughBehavior: "passthrough-behavior",
      payloadFormatVersion: "payload-format-version",
      templateSelectionExpression: "template-selection-expression",
      timeoutInMillis: 60,
      requestParameters: {
        "request-parameter": "request-parameter-value",
      },
      requestTemplates: {
        "request-template": "request-template-value",
      },
      responseParameters: {
        "200": {
          ResponseParameters: [
            {
              Source: "source",
              Destination: "destination",
            },
          ],
        },
      },
      tlsConfig: {
        serverNameToVerify: "server-name-to-verify",
      },
    },
    Apigatewayv2Integration,
    {
      apiId: "api-id",
      connectionId: "connection-id",
      connectionType: "connection-type",
      contentHandlingStrategy: "content-handling-strategy",
      credentialsArn: "credentials-arn",
      description: "description",
      integrationMethod: "integration-method",
      integrationSubtype: "integration-subtype",
      integrationType: "integration-type",
      integrationUri: "integration-uri",
      passthroughBehavior: "passthrough-behavior",
      payloadFormatVersion: "payload-format-version",
      templateSelectionExpression: "template-selection-expression",
      timeoutMilliseconds: 60,
      requestParameters: {
        "request-parameter": "request-parameter-value",
      },
      responseParameters: [
        {
          statusCode: "200",
          mappings: { source: "destination" },
        },
      ],
      requestTemplates: {
        "request-template": "request-template-value",
      },
      tlsConfig: {
        serverNameToVerify: "server-name-to-verify",
      },
    },
  );

  itShouldMapCfnElementToTerraformResource(
    CfnAuthorizerV2,
    {
      apiId: "api-id",
      authorizerCredentialsArn: "authorizer-credentials-arn",
      authorizerPayloadFormatVersion: "authorizer-payload-format-version",
      authorizerResultTtlInSeconds: 60,
      authorizerType: "authorizer-type",
      authorizerUri: "authorizer-uri",
      enableSimpleResponses: true,
      identitySource: ["identity-source"],
      name: "name",
      jwtConfiguration: {
        audience: ["audience"],
        issuer: "issuer",
      },
    },
    Apigatewayv2Authorizer,
    {
      apiId: "api-id",
      name: "name",
      jwtConfiguration: {
        audience: ["audience"],
        issuer: "issuer",
      },
      enableSimpleResponses: true,
      identitySources: ["identity-source"],
      authorizerCredentialsArn: "authorizer-credentials-arn",
      authorizerPayloadFormatVersion: "authorizer-payload-format-version",
      authorizerResultTtlInSeconds: 60,
      authorizerType: "authorizer-type",
      authorizerUri: "authorizer-uri",
    },
  );
});
