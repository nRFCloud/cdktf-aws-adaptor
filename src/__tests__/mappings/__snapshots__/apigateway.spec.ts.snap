// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Apigateway mappings > Should map AWS::ApiGatewayV2::Authorizer 1`] = `
"{
  \\"provider\\": {
    \\"aws\\": [
      {
        \\"alias\\": \\"us_east_1\\",
        \\"region\\": \\"us-east-1\\"
      }
    ]
  },
  \\"resource\\": {
    \\"aws_apigatewayv2_authorizer\\": {
      \\"resource_22C949BF\\": {
        \\"api_id\\": \\"api-id\\",
        \\"authorizer_credentials_arn\\": \\"authorizer-credentials-arn\\",
        \\"authorizer_payload_format_version\\": \\"authorizer-payload-format-version\\",
        \\"authorizer_result_ttl_in_seconds\\": 60,
        \\"authorizer_type\\": \\"authorizer-type\\",
        \\"authorizer_uri\\": \\"authorizer-uri\\",
        \\"enable_simple_responses\\": true,
        \\"identity_sources\\": [
          \\"identity-source\\"
        ],
        \\"jwt_configuration\\": {
          \\"audience\\": [
            \\"audience\\"
          ],
          \\"issuer\\": \\"issuer\\"
        },
        \\"name\\": \\"name\\"
      }
    }
  },
  \\"terraform\\": {
    \\"backend\\": {
      \\"local\\": {
        \\"path\\": \\"/terraform.test-stack.tfstate\\"
      }
    },
    \\"required_providers\\": {
      \\"aws\\": {
        \\"source\\": \\"aws\\",
        \\"version\\": \\"5.64.0\\"
      }
    }
  },
  \\"variable\\": {
    \\"resource_refs_CDA17697\\": {
      \\"default\\": \\"\${aws_apigatewayv2_authorizer.resource_22C949BF.id}\\",
      \\"type\\": \\"string\\"
    }
  }
}"
`;

exports[`Apigateway mappings > Should map AWS::ApiGatewayV2::DomainName 1`] = `
"{
  \\"provider\\": {
    \\"aws\\": [
      {
        \\"alias\\": \\"us_east_1\\",
        \\"region\\": \\"us-east-1\\"
      }
    ]
  },
  \\"resource\\": {
    \\"aws_apigatewayv2_domain_name\\": {
      \\"resource_22C949BF\\": {
        \\"domain_name\\": \\"domain-name\\",
        \\"domain_name_configuration\\": {
          \\"certificate_arn\\": \\"certificate-arn\\",
          \\"endpoint_type\\": \\"REGIONAL\\",
          \\"ownership_verification_certificate_arn\\": \\"ownership-verification-certificate-arn\\",
          \\"security_policy\\": \\"TLS_1_2\\"
        },
        \\"mutual_tls_authentication\\": {
          \\"truststore_uri\\": \\"truststore-uri\\",
          \\"truststore_version\\": \\"truststore-version\\"
        },
        \\"tags\\": {
          \\"tag\\": \\"tag-value\\"
        }
      }
    }
  },
  \\"terraform\\": {
    \\"backend\\": {
      \\"local\\": {
        \\"path\\": \\"/terraform.test-stack.tfstate\\"
      }
    },
    \\"required_providers\\": {
      \\"aws\\": {
        \\"source\\": \\"aws\\",
        \\"version\\": \\"5.64.0\\"
      }
    }
  },
  \\"variable\\": {
    \\"resource_refs_CDA17697\\": {
      \\"default\\": \\"\${join(\\\\\\"\\\\\\", [aws_apigatewayv2_domain_name.resource_22C949BF.domain_name_configuration[0].target_domain_name, \\\\\\",\\\\\\", aws_apigatewayv2_domain_name.resource_22C949BF.domain_name_configuration[0].hosted_zone_id])}\\",
      \\"type\\": \\"string\\"
    }
  }
}"
`;

exports[`Apigateway mappings > Should map AWS::ApiGatewayV2::Integration 1`] = `
"{
  \\"provider\\": {
    \\"aws\\": [
      {
        \\"alias\\": \\"us_east_1\\",
        \\"region\\": \\"us-east-1\\"
      }
    ]
  },
  \\"resource\\": {
    \\"aws_apigatewayv2_integration\\": {
      \\"resource_22C949BF\\": {
        \\"api_id\\": \\"api-id\\",
        \\"connection_id\\": \\"connection-id\\",
        \\"connection_type\\": \\"connection-type\\",
        \\"content_handling_strategy\\": \\"content-handling-strategy\\",
        \\"credentials_arn\\": \\"credentials-arn\\",
        \\"description\\": \\"description\\",
        \\"integration_method\\": \\"integration-method\\",
        \\"integration_subtype\\": \\"integration-subtype\\",
        \\"integration_type\\": \\"integration-type\\",
        \\"integration_uri\\": \\"integration-uri\\",
        \\"passthrough_behavior\\": \\"passthrough-behavior\\",
        \\"payload_format_version\\": \\"payload-format-version\\",
        \\"request_parameters\\": {
          \\"request-parameter\\": \\"request-parameter-value\\"
        },
        \\"request_templates\\": {
          \\"request-template\\": \\"request-template-value\\"
        },
        \\"response_parameters\\": [
          {
            \\"mappings\\": {
              \\"source\\": \\"destination\\"
            },
            \\"status_code\\": \\"200\\"
          }
        ],
        \\"template_selection_expression\\": \\"template-selection-expression\\",
        \\"timeout_milliseconds\\": 60,
        \\"tls_config\\": {
          \\"server_name_to_verify\\": \\"server-name-to-verify\\"
        }
      }
    }
  },
  \\"terraform\\": {
    \\"backend\\": {
      \\"local\\": {
        \\"path\\": \\"/terraform.test-stack.tfstate\\"
      }
    },
    \\"required_providers\\": {
      \\"aws\\": {
        \\"source\\": \\"aws\\",
        \\"version\\": \\"5.64.0\\"
      }
    }
  },
  \\"variable\\": {
    \\"resource_refs_CDA17697\\": {
      \\"default\\": \\"\${aws_apigatewayv2_integration.resource_22C949BF.id}\\",
      \\"type\\": \\"string\\"
    }
  }
}"
`;

exports[`Apigateway mappings > Should map AWS::ApiGatewayV2::Route 1`] = `
"{
  \\"provider\\": {
    \\"aws\\": [
      {
        \\"alias\\": \\"us_east_1\\",
        \\"region\\": \\"us-east-1\\"
      }
    ]
  },
  \\"resource\\": {
    \\"aws_apigatewayv2_route\\": {
      \\"resource_22C949BF\\": {
        \\"api_id\\": \\"api-id\\",
        \\"api_key_required\\": true,
        \\"authorization_scopes\\": [
          \\"authorization-scope\\"
        ],
        \\"authorization_type\\": \\"authorization-type\\",
        \\"authorizer_id\\": \\"authorizer-id\\",
        \\"model_selection_expression\\": \\"model-selection-expression\\",
        \\"operation_name\\": \\"operation-name\\",
        \\"request_models\\": {
          \\"request-model\\": \\"request-model-value\\"
        },
        \\"request_parameter\\": [
          {
            \\"request_parameter_key\\": \\"request-parameter\\",
            \\"required\\": false
          }
        ],
        \\"route_key\\": \\"route-key\\",
        \\"route_response_selection_expression\\": \\"route-response-selection-expression\\",
        \\"target\\": \\"target\\"
      }
    }
  },
  \\"terraform\\": {
    \\"backend\\": {
      \\"local\\": {
        \\"path\\": \\"/terraform.test-stack.tfstate\\"
      }
    },
    \\"required_providers\\": {
      \\"aws\\": {
        \\"source\\": \\"aws\\",
        \\"version\\": \\"5.64.0\\"
      }
    }
  },
  \\"variable\\": {
    \\"resource_refs_CDA17697\\": {
      \\"default\\": \\"\${aws_apigatewayv2_route.resource_22C949BF.id}\\",
      \\"type\\": \\"string\\"
    }
  }
}"
`;

exports[`Apigateway mappings > Should map AWS::ApiGatewayV2::Stage 1`] = `
"{
  \\"provider\\": {
    \\"aws\\": [
      {
        \\"alias\\": \\"us_east_1\\",
        \\"region\\": \\"us-east-1\\"
      }
    ]
  },
  \\"resource\\": {
    \\"aws_apigatewayv2_stage\\": {
      \\"resource_22C949BF\\": {
        \\"access_log_settings\\": {
          \\"destination_arn\\": \\"destination-arn\\",
          \\"format\\": \\"format\\"
        },
        \\"api_id\\": \\"api-id\\",
        \\"auto_deploy\\": true,
        \\"client_certificate_id\\": \\"client-certificate-id\\",
        \\"default_route_settings\\": {
          \\"data_trace_enabled\\": true,
          \\"detailed_metrics_enabled\\": true,
          \\"logging_level\\": \\"logging-level\\",
          \\"throttling_burst_limit\\": 60,
          \\"throttling_rate_limit\\": 60
        },
        \\"deployment_id\\": \\"deployment-id\\",
        \\"description\\": \\"description\\",
        \\"name\\": \\"stage-name\\",
        \\"route_settings\\": [
          {
            \\"data_trace_enabled\\": true,
            \\"route_key\\": \\"route-key\\"
          }
        ],
        \\"stage_variables\\": {
          \\"stage-variable\\": \\"stage-variable-value\\"
        },
        \\"tags\\": {
          \\"tag\\": \\"tag-value\\"
        }
      }
    }
  },
  \\"terraform\\": {
    \\"backend\\": {
      \\"local\\": {
        \\"path\\": \\"/terraform.test-stack.tfstate\\"
      }
    },
    \\"required_providers\\": {
      \\"aws\\": {
        \\"source\\": \\"aws\\",
        \\"version\\": \\"5.64.0\\"
      }
    }
  },
  \\"variable\\": {
    \\"resource_refs_CDA17697\\": {
      \\"default\\": \\"\${aws_apigatewayv2_stage.resource_22C949BF.id}\\",
      \\"type\\": \\"string\\"
    }
  }
}"
`;

exports[`Apigateway mappings > should map AWS::ApiGateway::Account 1`] = `
"{
  \\"provider\\": {
    \\"aws\\": [
      {
        \\"alias\\": \\"us_east_1\\",
        \\"region\\": \\"us-east-1\\"
      }
    ]
  },
  \\"resource\\": {
    \\"aws_api_gateway_account\\": {
      \\"resource_22C949BF\\": {
        \\"cloudwatch_role_arn\\": \\"cw-role-arn\\"
      }
    }
  },
  \\"terraform\\": {
    \\"backend\\": {
      \\"local\\": {
        \\"path\\": \\"/terraform.test-stack.tfstate\\"
      }
    },
    \\"required_providers\\": {
      \\"aws\\": {
        \\"source\\": \\"aws\\",
        \\"version\\": \\"5.64.0\\"
      }
    }
  },
  \\"variable\\": {
    \\"resource_refs_CDA17697\\": {
      \\"default\\": \\"\${aws_api_gateway_account.resource_22C949BF.id}\\",
      \\"type\\": \\"string\\"
    }
  }
}"
`;

exports[`Apigateway mappings > should map AWS::ApiGateway::BasePathMapping 1`] = `
"{
  \\"provider\\": {
    \\"aws\\": [
      {
        \\"alias\\": \\"us_east_1\\",
        \\"region\\": \\"us-east-1\\"
      }
    ]
  },
  \\"resource\\": {
    \\"aws_api_gateway_base_path_mapping\\": {
      \\"resource_22C949BF\\": {
        \\"api_id\\": \\"rest-api-id\\",
        \\"base_path\\": \\"base-path\\",
        \\"domain_name\\": \\"domain-name\\",
        \\"id\\": \\"id\\",
        \\"stage_name\\": \\"stage\\"
      }
    }
  },
  \\"terraform\\": {
    \\"backend\\": {
      \\"local\\": {
        \\"path\\": \\"/terraform.test-stack.tfstate\\"
      }
    },
    \\"required_providers\\": {
      \\"aws\\": {
        \\"source\\": \\"aws\\",
        \\"version\\": \\"5.64.0\\"
      }
    }
  },
  \\"variable\\": {
    \\"resource_refs_CDA17697\\": {
      \\"default\\": \\"\\",
      \\"type\\": \\"string\\"
    }
  }
}"
`;

exports[`Apigateway mappings > should map AWS::ApiGateway::Deployment 1`] = `
"{
  \\"provider\\": {
    \\"aws\\": [
      {
        \\"alias\\": \\"us_east_1\\",
        \\"region\\": \\"us-east-1\\"
      }
    ]
  },
  \\"resource\\": {
    \\"aws_cloudcontrolapi_resource\\": {
      \\"resource_22C949BF\\": {
        \\"desired_state\\": \\"{\\\\\\"DeploymentCanarySettings\\\\\\":{\\\\\\"PercentTraffic\\\\\\":50,\\\\\\"StageVariableOverrides\\\\\\":{\\\\\\"stage-variable\\\\\\":\\\\\\"stage-variable-value\\\\\\"},\\\\\\"UseStageCache\\\\\\":true},\\\\\\"Description\\\\\\":\\\\\\"description\\\\\\",\\\\\\"RestApiId\\\\\\":\\\\\\"rest-api-id\\\\\\",\\\\\\"StageDescription\\\\\\":{\\\\\\"AccessLogSetting\\\\\\":{\\\\\\"DestinationArn\\\\\\":\\\\\\"destination-arn\\\\\\"},\\\\\\"CacheClusterEnabled\\\\\\":true,\\\\\\"CacheClusterSize\\\\\\":\\\\\\"cache-cluster-size\\\\\\",\\\\\\"CacheDataEncrypted\\\\\\":true,\\\\\\"CacheTtlInSeconds\\\\\\":60,\\\\\\"CachingEnabled\\\\\\":true,\\\\\\"CanarySetting\\\\\\":{\\\\\\"PercentTraffic\\\\\\":50,\\\\\\"StageVariableOverrides\\\\\\":{\\\\\\"stage-variable\\\\\\":\\\\\\"stage-variable-value\\\\\\"}},\\\\\\"ClientCertificateId\\\\\\":\\\\\\"client-certificate-id\\\\\\",\\\\\\"DataTraceEnabled\\\\\\":true,\\\\\\"DocumentationVersion\\\\\\":\\\\\\"documentation-version\\\\\\",\\\\\\"LoggingLevel\\\\\\":\\\\\\"logging-level\\\\\\",\\\\\\"MethodSettings\\\\\\":[{\\\\\\"CacheDataEncrypted\\\\\\":true,\\\\\\"CacheTtlInSeconds\\\\\\":60,\\\\\\"CachingEnabled\\\\\\":true,\\\\\\"DataTraceEnabled\\\\\\":true,\\\\\\"HttpMethod\\\\\\":\\\\\\"http-method\\\\\\",\\\\\\"LoggingLevel\\\\\\":\\\\\\"logging-level\\\\\\",\\\\\\"MetricsEnabled\\\\\\":true,\\\\\\"ResourcePath\\\\\\":\\\\\\"resource-path\\\\\\",\\\\\\"ThrottlingBurstLimit\\\\\\":60,\\\\\\"ThrottlingRateLimit\\\\\\":60}],\\\\\\"MetricsEnabled\\\\\\":true,\\\\\\"ThrottlingBurstLimit\\\\\\":60,\\\\\\"ThrottlingRateLimit\\\\\\":60,\\\\\\"TracingEnabled\\\\\\":true},\\\\\\"StageName\\\\\\":\\\\\\"stage-name\\\\\\"}\\",
        \\"type_name\\": \\"AWS::ApiGateway::Deployment\\"
      }
    }
  },
  \\"terraform\\": {
    \\"backend\\": {
      \\"local\\": {
        \\"path\\": \\"/terraform.test-stack.tfstate\\"
      }
    },
    \\"required_providers\\": {
      \\"aws\\": {
        \\"source\\": \\"aws\\",
        \\"version\\": \\"5.64.0\\"
      }
    }
  },
  \\"variable\\": {
    \\"resource_refs_CDA17697\\": {
      \\"default\\": \\"\${jsondecode(aws_cloudcontrolapi_resource.resource_22C949BF.properties).DeploymentId}\\",
      \\"type\\": \\"string\\"
    }
  }
}"
`;

exports[`Apigateway mappings > should map AWS::ApiGateway::Stage 1`] = `
"{
  \\"provider\\": {
    \\"aws\\": [
      {
        \\"alias\\": \\"us_east_1\\",
        \\"region\\": \\"us-east-1\\"
      }
    ]
  },
  \\"resource\\": {
    \\"aws_cloudcontrolapi_resource\\": {
      \\"resource_22C949BF\\": {
        \\"desired_state\\": \\"{\\\\\\"AccessLogSetting\\\\\\":{\\\\\\"DestinationArn\\\\\\":\\\\\\"destination-arn\\\\\\"},\\\\\\"CacheClusterEnabled\\\\\\":true,\\\\\\"CacheClusterSize\\\\\\":\\\\\\"cache-cluster-size\\\\\\",\\\\\\"CanarySetting\\\\\\":{\\\\\\"PercentTraffic\\\\\\":50,\\\\\\"StageVariableOverrides\\\\\\":{\\\\\\"stage-variable\\\\\\":\\\\\\"stage-variable-value\\\\\\"}},\\\\\\"ClientCertificateId\\\\\\":\\\\\\"client-certificate-id\\\\\\",\\\\\\"DeploymentId\\\\\\":\\\\\\"deployment-id\\\\\\",\\\\\\"Description\\\\\\":\\\\\\"description\\\\\\",\\\\\\"DocumentationVersion\\\\\\":\\\\\\"documentation-version\\\\\\",\\\\\\"MethodSettings\\\\\\":[],\\\\\\"RestApiId\\\\\\":\\\\\\"rest-api-id\\\\\\",\\\\\\"StageName\\\\\\":\\\\\\"stage-name\\\\\\",\\\\\\"TracingEnabled\\\\\\":true}\\",
        \\"type_name\\": \\"AWS::ApiGateway::Stage\\"
      }
    }
  },
  \\"terraform\\": {
    \\"backend\\": {
      \\"local\\": {
        \\"path\\": \\"/terraform.test-stack.tfstate\\"
      }
    },
    \\"required_providers\\": {
      \\"aws\\": {
        \\"source\\": \\"aws\\",
        \\"version\\": \\"5.64.0\\"
      }
    }
  },
  \\"variable\\": {
    \\"resource_refs_CDA17697\\": {
      \\"default\\": \\"\\",
      \\"type\\": \\"string\\"
    }
  }
}"
`;
